ospf6d/ospf6_neighbor.c

```diff
@@ -59,7 +59,8 @@ int ospf6_neighbor_cmp(void *va, void *vb)
 {
 	struct ospf6_neighbor *ona = (struct ospf6_neighbor *)va;
 	struct ospf6_neighbor *onb = (struct ospf6_neighbor *)vb;
-	return (ntohl(ona->router_id) < ntohl(onb->router_id) ? -1 : 1);
+
+	return ntohl(ona->router_id) - ntohl(onb->router_id);
```

Comment:
This doesn't work.  Assume A is 128.0.0.2 (0x80000002) while B is 0.0.0.1.  Subtracting B from A gives 0x80000001, which is -2147483647 as a signed int.

(The signed difference between two 32-bit unsigned ints is a 33-bit value...)

Reply:
Very good catch. I remember hitting this exact problem in the past while working on ldpd, which prompted me to write the `ldp_addrcmp()` function like this: https://github.com/FRRouting/frr/blob/master/ldpd/util.c#L141-L144

Oh God... C is great, isn't it?

Reply:
fixing